ext/curl/tests/bug76675.phpt

```diff
@@ -30,7 +26,7 @@ $mh = curl_multi_init();
 curl_multi_setopt($mh, CURLMOPT_PIPELINING, CURLPIPE_MULTIPLEX);
 curl_multi_setopt($mh, CURLMOPT_PUSHFUNCTION, $callback);
 $ch = curl_init();
-curl_setopt($ch, CURLOPT_URL, 'https://http2.golang.org/serverpush');
+curl_setopt($ch, CURLOPT_URL, 'https://localhost/serverpush');
```

Comment:
All tests segfault for me, for the same reason. I don't know curl enough to understand what could be causing this.

```
AddressSanitizer:DEADLYSIGNAL
=================================================================
==2731264==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000000 (pc 0x7fc59f17d09d bp 0x7ffdf609c940 sp 0x7ffdf609c8a8 T0)
==2731264==The signal is caused by a WRITE memory access.
==2731264==Hint: address points to the zero page.
    #0 0x7fc59f17d09d in __memcpy_avx_unaligned_erms (/lib64/libc.so.6+0x15b09d) (BuildId: 245240a31888ad5c11bbc55b18e02d87388f59a9)
    #1 0x7fc5a00e9db0 in on_frame_recv (/lib64/libcurl.so.4+0x33db0) (BuildId: 354cfd307c3ca02d7c9c0386fc1a52d496689a0d)
    #2 0x7fc59e9de493 in session_after_header_block_received (/lib64/libnghttp2.so.14+0x9493) (BuildId: e10861ccd7fab76bc152927562f85635e27008f9)
    #3 0x7fc59e9e5634 in nghttp2_session_mem_recv (/lib64/libnghttp2.so.14+0x10634) (BuildId: e10861ccd7fab76bc152927562f85635e27008f9)
    #4 0x7fc5a00e221d in h2_process_pending_input.lto_priv.0 (/lib64/libcurl.so.4+0x2c21d) (BuildId: 354cfd307c3ca02d7c9c0386fc1a52d496689a0d)
    #5 0x7fc5a00e2a31 in cf_h2_recv.lto_priv.0 (/lib64/libcurl.so.4+0x2ca31) (BuildId: 354cfd307c3ca02d7c9c0386fc1a52d496689a0d)
    #6 0x7fc5a00f7f1d in Curl_read (/lib64/libcurl.so.4+0x41f1d) (BuildId: 354cfd307c3ca02d7c9c0386fc1a52d496689a0d)
    #7 0x7fc5a010bd09 in Curl_readwrite (/lib64/libcurl.so.4+0x55d09) (BuildId: 354cfd307c3ca02d7c9c0386fc1a52d496689a0d)
    #8 0x7fc5a0100602 in multi_runsingle (/lib64/libcurl.so.4+0x4a602) (BuildId: 354cfd307c3ca02d7c9c0386fc1a52d496689a0d)
    #9 0x7fc5a0102d8c in curl_multi_perform (/lib64/libcurl.so.4+0x4cd8c) (BuildId: 354cfd307c3ca02d7c9c0386fc1a52d496689a0d)
    #10 0xa67e19 in zif_curl_multi_exec /home/ilutov/Developer/php-src/ext/curl/multi.c:225
    #11 0x1762970 in ZEND_DO_ICALL_SPEC_RETVAL_USED_HANDLER /home/ilutov/Developer/php-src/Zend/zend_vm_execute.h:1297
    #12 0x19eb44e in execute_ex /home/ilutov/Developer/php-src/Zend/zend_vm_execute.h:55791
    #13 0x1a0ca74 in zend_execute /home/ilutov/Developer/php-src/Zend/zend_vm_execute.h:60163
    #14 0x165c083 in zend_execute_scripts /home/ilutov/Developer/php-src/Zend/zend.c:1846
    #15 0x14008f4 in php_execute_script /home/ilutov/Developer/php-src/main/main.c:2542
    #16 0x1daba45 in do_cli /home/ilutov/Developer/php-src/sapi/cli/php_cli.c:965
    #17 0x1daeb0d in main /home/ilutov/Developer/php-src/sapi/cli/php_cli.c:1367
    #18 0x7fc59f049b49 in __libc_start_call_main (/lib64/libc.so.6+0x27b49) (BuildId: 245240a31888ad5c11bbc55b18e02d87388f59a9)
    #19 0x7fc59f049c0a in __libc_start_main_alias_2 (/lib64/libc.so.6+0x27c0a) (BuildId: 245240a31888ad5c11bbc55b18e02d87388f59a9)
    #20 0x603fa4 in _start (/home/ilutov/Developer/php-src/sapi/cli/php+0x603fa4) (BuildId: aafeb2a13d8f05a7f10c1469375e68f2ff3ee0c5)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV (/lib64/libc.so.6+0x15b09d) (BuildId: 245240a31888ad5c11bbc55b18e02d87388f59a9) in __memcpy_avx_unaligned_erms
==2731264==ABORTING
```

Reply:
I now have the same issue (tests were green the last time I worked on this patch). That's why we need these tests :D

Reply:
I get a similar backtrace with GDB:

```

Reply:
I suspect some kind of race condition as these tests sometimes pass, and sometimes segfault. I'm taking a look but since I'm not a curl specialist either, I'm not promising anything 😅.

Reply:
I reproduced the problem using https://github.com/dunglas/php-dev-docker.
Upgrading to curl 8 fixes the issue, so it's likely a CURL bug that already been fixed.

Reply:
@dunglas Thanks for having a look! Sure, skipping the tests for the broken curl versions would be appropriate.

Reply:
@iluuu1994 done. This patch should be ready to be merged.